lib/create/attach-to-issue-from-message.js

```diff
@@ -0,0 +1,219 @@
+        addComment,
+        { id: issueId, body: comment },
+      ));
+    }
+  } catch (err) {
+    if (!(err.name && err.name === 'GraphQLError')) {
+      throw err;
+    }
+    req.log.debug({ err }, 'Could not create comment');
+
+    const appUrl = (await robot.info()).html_url;
+    return res.send({
+      errors: [{
+        name: 'issueId',
+        error: `Please make sure the Slack app has been granted issue and pull request write access on this repository: ${appUrl}`,
+      }],
+    });
+  }
+
+  let issueUrl;
+  if (data.html_url) {
+    issueUrl = data.html_url;
+  } else {
+    issueUrl = data.addComment.commentEdge.node.url;
+  }
+
+  await axios.post(response_url, {
+    response_type: 'ephemeral',
+    attachments: [new CommentCreated(issueUrl).getAttachment()],
+  });
+  return res.send();
+}
+
+async function loadIssuesAndPrs(req, res) {
+  function formatLabel(issue, includeRepository = false) {
+    const label = `#${issue.number} ${issue.title}`;
+    if (includeRepository) {
+      return `[${issue.repository.nameWithOwner}] ${label}`;
+    }
+    if (label.length > 30) {
+      return `${label.slice(0, 30).slice(0, -3)}...`;
+    }
+    return label;
+  }
+
+  const { slackUser } = res.locals;
+  const { value } = req.body;
+
+  const query = `
+    query {
+      viewer {
+        issues(first: 100, orderBy: {field: UPDATED_AT, direction: DESC}) {
+          nodes {
+            title
+            number
+            id
+            repository {
+              nameWithOwner
+            }
+          }
+        }
+      pullRequests(first: 100, orderBy: {field: UPDATED_AT, direction: DESC}) {
+        nodes {
+          title
+          number
+          id
+          repository {
+            nameWithOwner
+          }
+        }
+      }
+    }
+  }`;
+
+  const { data } = await slackUser.GitHubUser.client.request({
+    headers: {
+      accept: 'application/json',
+      'content-type': 'application/json',
+    },
+    method: 'POST',
+    url: '/graphql',
+    query,
+  });
+
+  const { issues, pullRequests } = data.data.viewer;
+  let rawIssues = issues.nodes.concat(pullRequests.nodes).filter(issue => issue);
+
+  // If user supplied search term, then we filter based on the search term
+  if (value) {
+    rawIssues = rawIssues.filter(issue => formatLabel(issue, true).toLowerCase().includes(value));
+  }
+
+  // Slack permits a maximum of 100 options
+  rawIssues = rawIssues.slice(0, 100);
+
+  // Re-order all issues and PRs by repository for grouping in the dialog
+  const issuesByRepo = {};
+  rawIssues.forEach((issue) => {
+    if (issuesByRepo[issue.repository.nameWithOwner]) {
+      issuesByRepo[issue.repository.nameWithOwner].push({
+        label: formatLabel(issue),
+        value: issue.id,
+      });
+    } else {
+      issuesByRepo[issue.repository.nameWithOwner] = [
+        {
+          label: formatLabel(issue),
+          value: issue.id,
+        },
+      ];
+    }
+  });
```

Comment:
I would be great to extract all of this manipulation into a module/model of some sort.